File: files/2024-06-02/CSEC 02.06.R
library(tidyverse)
library(readxl)

input = read_excel("files/Excel Challenge 2nd June.xlsx", range = "C2:C8")
test  = read_excel("files/Excel Challenge 2nd June.xlsx", range = "D2:D8")

result = input %>%
  mutate(last_letter = map_int(str_locate_all(`Customers & Orders`,
                                              "[A-Za-z]"),
                               ~max(.x[,2]))) %>%
  mutate(answer = ifelse(last_letter < nchar(`Customers & Orders`), 
                         str_sub(`Customers & Orders`, last_letter + 1), 
                         NA_character_))
           
identical(result$answer, test$`Last Correct Order`)         
#> [1] TRUE

File: files/2024-06-02/CSEC 02.06.py
import pandas as pd
import re

path = "Excel Challenge 2nd June.xlsx"
input_data = pd.read_excel(path, usecols="C", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=7)

def tail_after_last_letter(text):
    matches = list(re.finditer(r"[A-Za-z]", str(text)))
    if not matches:
        return None
    last_pos = matches[-1].end()
    return text[last_pos:] if last_pos < len(text) else None

result = input_data.assign(answer=input_data["Customers & Orders"].map(tail_after_last_letter))
print(result["answer"].equals(test["Last Correct Order"]))
